blackjax/mcmc/trajectory.py

```diff
@@ -625,6 +625,11 @@ def update_sum_log_p_accept(inputs):
 
 def hmc_energy(kinetic_energy):
     def energy(state):
-        return -state.logdensity + kinetic_energy(state.momentum)
+        # TODO: This is extremely inelegant!
```

Comment:
LOL yikes.
Let's change the kinetic_energy implementation to:
```
def kinetic_energy(momentum, position=None):
    ...
```
and for `gaussian_euclidean` we just `del position`

Reply:
This is a very sensible suggestion!! 😄